src/probe.cpp

```diff
@@ -0,0 +1,208 @@
+/* OpenAWE - A reimplementation of Remedys Alan Wake Engine
+ *
+ * OpenAWE is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * OpenAWE is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * OpenAWE is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with OpenAWE. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <spdlog/spdlog.h>
+
+#include "src/awe/resman.h"
+#include "src/awe/rmdparchive.h"
+#include "src/common/exception.h"
+#include "src/probe.h"
+
+Probe::Probe() {}
+
+ProbeResult Probe::performProbe() {
+	spdlog::info("Probing archives...");
+	uint8_t allChecks = 
```

Comment:
Since `allChecks` combines results of all checks, it may receive a value outside of ones defined in `ProbeResult` enum - tried to compile it, got an error. I'll switch to using `std::bitset` instead to make the purpose of the variable clearer.